qtranslate_core.php

```diff
@@ -99,7 +99,7 @@ function qtranxf_init_language() {
 			 */
 			$target = apply_filters( 'qtranslate_language_detect_redirect', $url_lang, $url_orig, $url_info );
 			//qtranxf_dbg_log('qtranxf_init_language: doredirect to '.$lang.PHP_EOL .'urlorg:'.$url_orig.PHP_EOL .'target:'.$target.PHP_EOL .'url_info: ',$url_info);
-			if ( $target !== false && $target != $url_orig ) {
+			if( $target !== false && $target != $url_orig && ! is_rest_api_request( $url_orig ) ) {
```

Comment:
See the changes i made in the language detection to prevent redirects (qTranslate-XT 3.5.3). With the new rewrite rules there should not be any need to check later, because the final part of the URL is expected to be the same so the new .